capstone/capdb/models.py

```diff
@@ -421,13 +458,153 @@ class CaseXML(BaseXMLModel):
     case_id = models.CharField(max_length=255, db_index=True)
 
     metadata = models.OneToOneField(CaseMetadata, blank=True, null=True, related_name='case_xml',
-                                     on_delete=models.SET_NULL)
+                                    on_delete=models.SET_NULL)
     volume = models.ForeignKey(VolumeXML, related_name='case_xmls',
-                                     on_delete=models.DO_NOTHING)
+                               on_delete=models.DO_NOTHING)
     s3_key = models.CharField(max_length=1024, blank=True, help_text="s3 path")
 
     tracker = FieldTracker()
 
+    def save(self, force_insert=False, force_update=False, *args, **kwargs):
+
+        # This compares new XML to the old and updates ALTO files as necessary.
+        # We don't yet support adding or removing words from the case body.
+        if self.tracker.has_changed('orig_xml') and self.pk is not None and self.tracker.changed()['orig_xml']:
+            def __in_casebody(element):
+                return element.tag.startswith("{" + nsmap['casebody']) and not element.tag.endswith('casebody')
+
+            # parse objects and get the xml tree objects
+            parsed_orig_case = parse_xml(self.tracker.changed()['orig_xml'])
+            parsed_updated_case = parse_xml(self.orig_xml)
+            updated_tree = parsed_updated_case.root
+            original_tree = parsed_orig_case.root
+
+            # If the case is duplicative, just save and move on
+            if updated_tree.find('//duplicative:casebody', nsmap) is not None:
+                super(CaseXML, self).save(force_insert, force_update, *args, **kwargs)
+                return
+
+            if self.case_id == '':
```

Comment:
I think we're headed for just keeping CaseMetadata.case_id, which should always be set. Maybe just drop this?